File: IT/Strutture dati/Esercizi/Java/Esercizio27.java
/* file Esercizio27.java */

import java.util.Arrays;
import java.util.Random;

public class Esercizio27 {
	public static void main(String [] args) {
		Random rand = new Random();
		int [] nums = new int[100];

		for (int i = 0; i < nums.length; i++) {
			nums[i] = rand.nextInt(100); // numero casuale da 0 a 99
		}

		System.out.println(Arrays.toString(nums));

		int md = 1;
		int [] counts = new int[nums.length];

		for (int i = 0; i < counts.length; i++) {
			int c = nums[i];
			counts[c]++;
			if (counts[c] > md) {
				md = counts[c];
			}
		}

		if (md == 1) {
			System.out.println("L'array non presenta una moda.");
			return;
		}

		System.out.print("Moda (" + md + " occorrenze): ");

		int j = 0;
		for (int i = 0; i < counts.length; i++) {
			if (counts[i] == md) {
				if (j > 0) System.out.print(", ");
				System.out.print(i);
				j++;
			}
		}

		System.out.print(System.lineSeparator());
	}
}

File: IT/Strutture dati/Esercizi/Python/esercizio27.py
# file esercizio27.py

from __future__ import print_function
from random import randint
from sys import stdout
from os import linesep

nums = [randint(0, 99) for n in range(100)]

print(nums)

counts = [0] * len(nums)
md = 1

for i in range(0, len(nums)):
	c = nums[i]
	counts[c] += 1
	if counts[c] > md:
		md = counts[c]

if md == 1:
	print("La lista non ha una moda.")
	exit()

print("Moda ({} occorrenze): ".format(md), end="")

j = 0
for i in range(0, len(counts)):
	if counts[i] == md:
		if j > 0: print(", ", end="")
		print(i, end="")
		j += 1

print(".")

